cleanlab/datalab/internal/issue_manager/underperf_group.py

```diff
@@ -0,0 +1,281 @@
+    HIGHEST_PERFORMANCE = 1
+
+    def __init__(
+        self,
+        datalab: Datalab,
+        metric: Optional[str] = None,
+        threshold: float = 0.1,
+        **_,
+    ):
+        super().__init__(datalab)
+        self.metric = metric
+        self.threshold = self._set_threshold(threshold)
+
+    def find_issues(
+        self,
+        features: npt.NDArray,
+        pred_probs: npt.NDArray,
+        **kwargs,
+    ) -> None:
+        labels = self.datalab.labels
+        knn_graph = self.set_knn_graph(features, kwargs)
+        unique_cluster_labels, cluster_labels, cluster_obj = self.perform_clustering(knn_graph)
+        if not unique_cluster_labels.size:
+            raise ValueError(
+                "No meaningful clusters were generated for determining underperforming group."
+            )
+        n_clusters = len(unique_cluster_labels)
+        worst_cluster_id, worst_cluster_ratio = self.get_worst_cluster(
+            cluster_labels, unique_cluster_labels, labels, pred_probs
+        )
+        is_issue_column = cluster_labels == worst_cluster_id
+        scores = np.where(is_issue_column, worst_cluster_ratio, 1)
+        self.issues = pd.DataFrame(
+            {
+                f"is_{self.issue_name}_issue": is_issue_column,
+                self.issue_score_key: scores,
+            },
+        )
+        self.summary = self.make_summary(score=worst_cluster_ratio)
+        self.info = self.collect_info(
+            knn_graph=knn_graph,
+            n_clusters=n_clusters,
+            cluster_labels=cluster_labels,
+            cluster_obj=cluster_obj,
+        )
+
+    def set_knn_graph(self, features: npt.NDArray, find_issues_kwargs: Dict) -> csr_matrix:
+        knn_graph = self._process_knn_graph_from_inputs(find_issues_kwargs)
+        old_knn_metric = self.datalab.get_info("statistics").get("knn_metric")
+        metric_changes = self.metric and self.metric != old_knn_metric
+
+        if knn_graph is None or metric_changes:
+            if features is None:
+                raise ValueError(
+                    "If a knn_graph is not provided, features must be provided to fit a new knn."
+                )
+            if self.metric is None:
+                self.metric = "cosine" if features.shape[1] > 3 else "euclidean"
+            knn = NearestNeighbors(metric=self.metric)
+
+            if self.metric and self.metric != knn.metric:
+                warnings.warn(
+                    f"Metric {self.metric} does not match metric {knn.metric} used to fit knn. "
+                    "Most likely an existing NearestNeighbors object was passed in, but a different "
+                    "metric was specified."
+                )
+            self.metric = knn.metric
+
+            try:
+                check_is_fitted(knn)
+            except NotFittedError:
+                knn.fit(features)
+
+            knn_graph = knn.kneighbors_graph(mode="distance", n_neighbors=features.shape[0] - 1)
```

Comment:
Make sure we use a sensible default value of `k` here. Computing the full-connectivity is not sensible for large datasets.

Reply:
the default should just be the same as for the other issue types right? I don't think there should be a bunch of different `K` defaults

Reply:
Agreed, `k=10` is the consistent default we've set across other issue types. It's a reasonable starting point for this context as well.

If our experiments show significant improvements in clustering accuracy with a higher value, we can consider adjusting it.

Reply:
Some context  here - I had initially computed the knn graph with `k` as the number of data points, since HDBSCAN expects a symmetric distance matrix when `metric="precomputed"`.  Since this is computationally expensive, we have discussed to try and create a symmetric matrix using a smaller value of `k`, and as @elisno mentioned, experiments with different `k` can provide a better idea regarding clustering performance.